lib/storage/date_metric_id_cache.go

```diff
@@ -46,23 +60,40 @@ func (dmc *dateMetricIDCache) resetLocked() {
 }
 
 func (dmc *dateMetricIDCache) EntriesCount() int {
-	byDate := dmc.byDate.Load()
+	dmc.mu.Lock()
+	defer dmc.mu.Unlock()
+
 	n := 0
-	for _, metricIDs := range byDate.m {
+	for _, metricIDs := range dmc.byDate.Load().m {
+		n += metricIDs.Len()
+	}
+	for _, metricIDs := range dmc.byDateMutable.m {
 		n += metricIDs.Len()
 	}
 	return n
 }
 
 func (dmc *dateMetricIDCache) SizeBytes() uint64 {
```

Comment:
WDYT of create a method that exports stats - 

```func (dmc *dateMetricIDCache) UpdateStats(dst *cacheStats)  {
    dmc.mu.Lock()
	defer dmc.mu.Unlock()
    var totalSize int
    var entriesCount int
    
	for _, metricIDs := range dmc.byDate.Load().m {
		entriesCount += metricIDs.Len()
        totalSize += metricIDs.SizeBytes()
	}
	for _, metricIDs := range dmc.byDateMutable.m {
		entriesCount += metricIDs.Len()
        totalSize += metricIDs.SizeBytes()
	}
}```

 It will acquire mutex only once and export consistent view ( if cache changes during mutex.lock, metrics will be the same).